File: sdk/python/src/oathmesh/redis.py
from typing import Any
from .types import RevocationList

class RedisRevocationCache(RevocationList):
    """Redis-backed revocation cache implementation."""
    
    def __init__(self, client: Any, prefix: str = "om:rev:"):
        """
        Args:
            client: A redis-py client instance.
            prefix: Key prefix for redis.
        """
        self._client = client
        self._prefix = prefix

    def is_revoked(self, subject: str) -> bool:
        return self._client.get(f"{self._prefix}{subject}") is not None

File: sdk/node/src/redis.ts
import type { RevocationList } from './types';

/**
 * Redis client interface expected by the cache (compatible with node-redis).
 */
export interface RedisLike {
  get(key: string): Promise<string | null>;
}

export class RedisRevocationCache implements RevocationList {
  private client: RedisLike;
  private prefix: string;

  constructor(client: RedisLike, prefix: string = 'om:rev:') {
    this.client = client;
    this.prefix = prefix;
  }

  async isRevoked(subject: string): Promise<boolean> {
    const val = await this.client.get(`${this.prefix}${subject}`);
    return val !== null;
  }
}
